utils/tim_delay.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * tim_delay.h
```

Comment:
See comments in the `.c` file for details on the changes that should trickle over to the `.h`.